src/presenters/Whip_WPMessagePresenter.php

```diff
@@ -19,20 +19,33 @@ public function __construct( Whip_Message $message ) {
 	/**
 	 * Registers hooks to WordPress. This is a separate function so you can
 	 * control when the hooks are registered.
+	 *
+	 * @param Whip_MessageDismisser $dismisser Dismisser object.
 	 */
-	public function register_hooks() {
+	public function register_hooks( Whip_MessageDismisser $dismisser ) {
 		global $whip_admin_notices_added;
+
 		if ( null === $whip_admin_notices_added || ! $whip_admin_notices_added ) {
 			add_action( 'admin_notices', array( $this, 'renderMessage' ) );
 			$whip_admin_notices_added = true;
 		}
+
+		$dismissListener = new Whip_WPMessageDismissListener( $dismisser );
+		$dismissListener->listen();
 	}
 
 	/**
 	 * Renders the messages present in the global to notices.
 	 */
 	public function renderMessage() {
-	    printf( '<div class="error">%s</div>', $this->kses( $this->message->body() ) );
+		/* translators: 1: is a link to dismiss url 2: closing link tag */
+		$dismiss_button = sprintf(
+			__( '<p>%1$sRemind me again after the next WordPress release.%2$s</p>', 'wordpress' ),
+			'<a href="' . Whip_WPMessageDismissListener::get_dismissurl() . '">',
```

Comment:
As we have access to the actual instance of `$dismissListener` now, we can get rid of the static access here.